### DISCUSSION TITLE
JiraTestManager fails: how to circumvent it

### DISCUSSION THREAD

**Discussion by codectl** [pycontribs/jira#1485] [Ideas]
Probably everybody knows by now that the ```jira_ci``` workflow is failing on every PR due to the Jira cloud tests. The reason is that the cloud workflow contains secrets that are not configured and shared with the PR.

How can this be solved then?

**Comment by codectl** [comment]
One solution could be allowing the share of secrets with the PR. Since this can be a security concern, it's possible to block the execution of the pull request by the approval of one of the reviewers. Hence, the reviewer can check that the PR is safe to proceed, the cloud workflow runs and assess whether the code is safe to be merged with the base branch.

**Reply by adehad** [reply]
Unfortunately this is a limitation with GitHub Action itself.

If you know how to configure what you have described that would be great to share, we tried this earlier in the year and had limited success in the GitHub ecosystem.

I attempted something with Azure Pipelines that worked, but the added admin burden of managing two systems didn't appeal to me unfortunately.

**Reply by codectl** [reply]
Great , if you're receptive to the idea then I can put something together to show proof how to safely share secrets with a PR so that it doesn't break constantly on the cloud workflow . It would be an improvement over the creation of an extra branch just to run the cloud test suite.

**Reply by adehad** [reply]
If you have the time that would be great !
